File: src/main/java/dev/blachut/svelte/lang/SvelteHTMLLanguage.java
package dev.blachut.svelte.lang;

import com.intellij.lang.html.HTMLLanguage;
import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;

/**
 * Externally it's name is just "Svelte" because it's the root language of Svelte files
 */
public final class SvelteHTMLLanguage extends HTMLLanguage {
  public static final SvelteHTMLLanguage INSTANCE = new SvelteHTMLLanguage();

  private SvelteHTMLLanguage() {
    super(HTMLLanguage.INSTANCE, "SvelteHTML");
  }

  @Override
  public @NotNull String getDisplayName() {
    return "Svelte";
  }

  @Override
  public boolean isCaseSensitive() {
    return true;
  }

  @Override
  public @NotNull LanguageFileType getAssociatedFileType() {
    return SvelteHtmlFileType.INSTANCE;
  }
}

File: src/main/java/dev/blachut/svelte/lang/SvelteHTMLLanguage.kt
package dev.blachut.svelte.lang

import com.intellij.javascript.web.lang.html.WebFrameworkHtmlDialect

/**
 * Externally it's name is just "Svelte" because it's the root language of Svelte files
 */
class SvelteHTMLLanguage private constructor() : WebFrameworkHtmlDialect("SvelteHTML") {

    override fun isCaseSensitive(): Boolean = true

    companion object {
        @JvmField
        val INSTANCE = SvelteHTMLLanguage()
    }
}
